### PR TITLE
ci: make contract tests stage optional

### PR DESCRIPTION
**Pull request by jakubgs**
Contract tests are far too slow.

### CODE REVIEW

**Review by siddarthkay — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -375,25 +375,25 @@ test-contract: export SHADOW_NS_REGEXP := ^tests\.contract-test.*$$
...
-component-test: export TARGET := clojure
-component-test: export COMPONENT_TEST := true
-component-test: export BABEL_ENV := test
-component-test: export JEST_USE_SILENT_REPORTER := false
-component-test: ##@test Run component tests once in NodeJS
+test-component: export TARGET := clojure
+test-component: export COMPONENT_TEST := true
+test-component: export BABEL_ENV := test
+test-component: export JEST_USE_SILENT_REPORTER := false
+test-component: ##@test Run component tests once in NodeJS
```
Comment (J-Son89) [inline Makefile:396]: 🙏

File: ci/Jenkinsfile.tests [hunk 2]
```diff
@@ -29,6 +29,11 @@ pipeline {
       description: 'Specify build type. Values: pr / e2e / nightly / release',
       defaultValue: 'pr',
     )
+    booleanParam(
+      name: 'RUN_CONTRACT_TESTS',
+      description: 'Whether to run optional and slow contract tests.',
```
Comment (J-Son89) [inline ci/Jenkinsfile.tests:34]: are they slow or just timing out? 🤔

**Review by J-Son89 — APPROVED** [review]

File: ci/Jenkinsfile.tests (hunk 2 above)
Comment (siddarthkay) [reply ci/Jenkinsfile.tests:34]: I might even call them flaky.

Comment (jakubgs) [reply ci/Jenkinsfile.tests:34]: Those options are not mutually exclusive. Also, no idea.

**Pull request merged**